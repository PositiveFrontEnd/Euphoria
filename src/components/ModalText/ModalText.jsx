import React from "react";
import Modal from "../Modal/Modal";
import ModalBody from "../Modal/ModalBody";
import ModalFooter from "../Modal/ModalFooter";
import ModalWrapper from "../Modal/ModalWrapper";
import ModalClose from "../Modal/ModalClose";
import "../Modal/Modal.scss";
import PropTypes from "prop-types";
import ModalHeader from "../Modal/ModalHeader";
import { useSelector } from "react-redux";
import { selectorIsModal } from "../../store/selectors";
const ModalText = ({ data, handleClose, handleOK }) => {
  const { article, color, name, price, url } = data;
  const isModal = useSelector(selectorIsModal);
  return (
    <ModalWrapper isOpen={isModal} clickWrapper={handleClose}>
      <Modal>
        <ModalClose firstColor click={handleClose} />
        <ModalHeader>
          <img className="box__img" src={url} alt="picture" />
        </ModalHeader>
        <ModalBody>
          <div className="box__info">
            <h4 className="box__name">Product Name : {name} </h4>
            <p className="box__price">Price : {price} â‚´ </p>
            <span className="box__article">Product Code : {article} </span>
            <p className="box__color">Product Color : {color} </p>
          </div>
        </ModalBody>
        <ModalFooter
          buttonClassNameFirst="button__white"
          secondBox
          textFirst="Add TO Basket"
          clickFirst={handleOK}
        />
      </Modal>
    </ModalWrapper>
  );
};
ModalText.propTypes = {
  handleClose: PropTypes.func,
  isOpen: PropTypes.bool,
  data: PropTypes.object,
  handleOK: PropTypes.func,
};
export default ModalText;
