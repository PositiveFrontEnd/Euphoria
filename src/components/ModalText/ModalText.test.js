import React from "react";
import { fireEvent, getByText, render, screen } from "@testing-library/react";
import ModalText from "./ModalText";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
const handelClick = jest.fn();
describe("test Modal", () => {
  test("render Modal", () => {
    const mockStore = configureStore();
    const initialState = {
      favorites: [],
      cardsArr: [],
      favoritesIndex: [],
      baskets: [],
      basketsItem: [],
      currentCard: {},
      basketFormValue: {},
      basketsCount: [],
      isModal: false,
      isForm: false,
      isFormCurrentCurd: false,
    };
    const data = {
      article: 1,
      color: "red",
      name: "Yevhen",
      price: 200,
      url: "/",
    };
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ModalText data={data} />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
