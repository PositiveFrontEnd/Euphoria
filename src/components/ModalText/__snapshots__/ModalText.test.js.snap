// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Modal render Modal 1`] = `<DocumentFragment />`;
